src/lib/KeyShares/KeyShares.ts

```diff
@@ -71,6 +67,42 @@ export class KeyShares {
     return this.payload;
   }
 
+  generateKeySharesFromBytes(shares: string, operatorIds: any[]): any {
+    const operatorCount = operatorIds.length;
+    shares = shares.replace('0x', '');
+    const pkLength = parseInt(shares.substring(0, 4), 16);
+
+    // get the public keys part
+    const pkSplit = shares.substring(4, pkLength + 2);
+    const pkArray = ethers.utils.arrayify('0x' + pkSplit);
+    const sharesPublicKeys = this.splitArray(operatorCount, pkArray).map(item =>
+      ethers.utils.hexlify(item),
+    );
+
+    const eSplit = shares.substring(pkLength + 2);
+    const eArray = ethers.utils.arrayify('0x' + eSplit);
+    const encryptedKeys = this.splitArray(operatorCount, eArray).map(item =>
+      Buffer.from(ethers.utils.hexlify(item).replace('0x', ''), 'hex').toString(
+        'base64',
+      ),
+    );
+    return {
+      sharesPublicKeys,
+      encryptedKeys,
+    };
+  }
```

Comment:
I want us to stop using first bytes as length criteria.
Because we receive operator IDs and we know public key length we can do:

```
// One public key length is always 96
const publicKeyLength := 96

// All public keys length calculated as:
pubKeysLength := operatorIds.length * publicKeyLength

// Now you can just split by the length and have 2 parts: with public keys and encrypted keys.
// To get array of public keys and array of encrypted keys - you need to split each of them to equal parts with size of each as operatorIds.length
```

The second issue with this approach is that all the shares already by the fact are bytes, and we need to join/split them only, without any mutations as much as possible.

`bytes(pubKey1) + bytes(pubKey2) .... + bytes(encrytedKey1) + bytes(encryptedKey2) + ...` -> `single share`
With no 0x, with no length leading bytes etc.

And method to do opposite.